lib/kvdb/kvdb_cparams.c

```diff
@@ -263,8 +265,41 @@ kvdb_cparams_defaults()
     return params;
 }
 
+static merr_t
+kvdb_storage_paths_default_set(struct kvdb_cparams *params, const char *home, bool *pmem_only)
+{
+    merr_t err = 0;
+    bool   isdax;
+
+    if (pmem_only)
+        *pmem_only = false;
+
+    err = dax_path_is_fsdax(home, &isdax);
+    if (err)
+        return err;
+
+    if (isdax) {
+        /* If the pmem mclass path is not provided, then use the default path. */
+        if (params->storage.mclass[MP_MED_PMEM].path[0] == '\0')
+            strlcpy(params->storage.mclass[MP_MED_PMEM].path, MPOOL_PMEM_MCLASS_DEFAULT_PATH,
+                    sizeof(params->storage.mclass[MP_MED_PMEM].path));
+
+        if (pmem_only && params->storage.mclass[MP_MED_CAPACITY].path[0] == '\0' &&
+            params->storage.mclass[MP_MED_STAGING].path[0] == '\0')
+            *pmem_only = true;
```

Comment:
I would say I am not a huge fan of this logic being pushed into the resolve() function. I would rather ikvdb_create() determine for itself whether it is a pmem-only kvdb if possible. I would like for this function to continue to be stateless.

Reply:
Alright, I can move kvdb_storage_paths_default_set() into ikvdb.c and have hse_kvdb_create() call it prior to calling kvdb_cparams_resolve().

Reply:
I agree.  Unrelated comment:  wouldn't path_copy be more appropriately named path_join ?

Reply:
@tristan957, your thoughts on s/path_copy/path_join?

Reply:
Done.